src/backend/interfaces/NATS/py.client/caliopen_nats/delivery.py

```diff
@@ -59,18 +99,55 @@ def process_raw(self, raw_msg_id):
             raise NotFound
         log.debug('Retrieved raw message {}'.format(raw_msg_id))
 
-        qualifier = UserDMQualifier(self.user)
+        qualifier = UserDMQualifier(self.user, self.identity)
         message = qualifier.process_inbound(raw)
 
+        # before storing a new message,
+        # check if this external message has been already imported
+        external_refs = ModelMessageExternalRefLookup.filter(
+            user_id=self.user.user_id,
+            external_msg_id=message.external_msg_id)
+        if external_refs:
```

Comment:
There is code duplication between UserMessageDelivery and UserTwitterDMDelivery for management of MessageExternalRefLookup.

These 2 classes inherit from object, may we can create a parent class having this currently duplicated logic only in one place.

Reply:
refactored in commit #36a26a7a